services/py-api/src/app_factory.py

```diff
@@ -1,32 +1,100 @@
 from contextlib import asynccontextmanager
-from typing import AsyncIterator
+from typing import AsyncIterator, TypedDict
 
 from fastapi import FastAPI
+from motor.motor_asyncio import AsyncIOMotorClient
 
 from src.database.db_clients import mongo_db_client_provider
-from src.database.mongo.db_manager import mongo_db_manager_provider
-from src.server.routes.routes import Routes
+from src.database.mongo.db_manager import (
+    MongoDatabaseManager,
+    PARTICIPANTS_COLLECTION,
+    TEAMS_COLLECTION,
+    FEATURE_SWITCH_COLLECTION,
+)
+from src.database.mongo.transaction_manager import MongoTransactionManager
+from src.database.repository.feature_switch_repository import FeatureSwitchRepository
+from src.database.repository.hackathon.participants_repository import ParticipantsRepository
+from src.database.repository.hackathon.teams_repository import TeamsRepository
+from src.server.handlers.feature_switch_handler import FeatureSwitchHandler
+from src.server.handlers.hackathon.hackathon_handlers import HackathonManagementHandlers
+from src.server.handlers.hackathon.participants_handlers import ParticipantHandlers
+from src.server.handlers.hackathon.verification_handlers import VerificationHandlers
+from src.server.handlers.http_handlers import HttpHandlersContainer
+from src.server.handlers.utility_hanlders import UtilityHandlers
 from src.server.middleware.middleware import Middleware
+from src.server.routes.routes import Routes
+from src.service.feature_switch_service import FeatureSwitchService
+from src.service.hackathon.hackathon_mail_service import HackathonMailService
+from src.service.hackathon.hackathon_service import HackathonService
+from src.service.hackathon.participants_registration_service import ParticipantRegistrationService
+from src.service.hackathon.participants_verification_service import ParticipantVerificationService
+from src.service.jwt_utils.codec import JwtUtility
+from src.service.mail_service.mail_clients.mail_client_factory import mail_client_factory, MailClients
+
+
+class State(TypedDict):
+    db_client: AsyncIOMotorClient
 
 
 @asynccontextmanager
 async def lifespan(app: FastAPI) -> AsyncIterator[None]:
     """
     To learn more about lifespan events, visit:
     https://fastapi.tiangolo.com/advanced/events/#lifespan
+    https://www.starlette.io/lifespan/
+    https://asgi.readthedocs.io/en/latest/specs/lifespan.html
 
     To learn more about context managers, visit:
     https://docs.python.org/3/library/contextlib.html#contextlib.asynccontextmanager
     """
+    # Open a connection to Mongo
+    db_client = mongo_db_client_provider()
+
     yield
-    # We ignore the error as we know that the client is initialized
-    _ = mongo_db_manager_provider(mongo_db_client_provider()).close_all_connections()
+
+    # Close the opened connections towards MongoDB
+    db_client.close()
```

Comment:
I know this is straightforward for you but I suggest you throw a quick comment explaining when is the yield invoked and what it does